### DISCUSSION TITLE
could vpm be inspired by crates .io ?

### DISCUSSION THREAD

**Discussion by ffred** [vlang/v#18520] [Ideas]
Hi,
v is not yet as mature as Rust, but it would be nice if vpm could take some inspiration from crates.io..
actually there's no description on nearly every package, some point to error404 repository or even some point to repository for a different language !?
don't need to search for long. actually just on the first page for "new packages" :
"christopherzimmerman.vnum" point to error404 repository.
"Omega0x013.loadingbar" point to a Rust repository.
and there is no description for any package.
another odd thing, "Delta456.isthirt" is on the "new packages" page, but when I go to the github repository, everything is from "3 years ago" ??
and I just looked at a few of them randomly....

the package names are not really explicit, so at least a small descriptive sentence should be mandatory to publish a package I think.
and will there be some version numbering for v packages ?

**Comment by Wajinn** [comment]
It's not exactly fair, to compare crates.io at this time, because of their much different ages (Rust is far older) and then there is the huge organization that made and pushed it, Mozilla.  Vlang is a much more contributor centric, and "for the people" grassroots effort, which is clearly gaining ground and maturing.

The link that deals with VPM (more specifically), is "https://github.com/vlang/vpm".  There we can see that some of these issues have been brought up:

**1)** 404 repository

[Issue 103](https://github.com/vlang/vpm/issues/103) kind of deals with this.  It looks like some packages turned private, after they were submitted.  As xy3 suggested, looks like a periodic check needs to be made to remove packages that return a 404.

**2)** Trolling?

In the case of Omega0x013.loadingbar, where the person submitted an outright Rust package, it looks like trolling.  Preventative measures will have to be in place to stop vandals or rivals from attempting to do obvious silliness.  While the V community is more tolerant and respectful of others, clearly many competitors and their users are not (and are even encouraged to be hostile), so seek to do harm.

**3)** Age can be tricky

In some cases, like Delta456.isthirt, it might be trickier than it looks.  For instance, somebody might have an old V code repository with something they think is good, that they suddenly submitted to VPM.  The age of the package would likely reflect the submission date, not the age of the repository.  Seems like some additional data will have to be pulled, like last commit date, then added to the description.

**4)** Corrections to site hopefully coming

Edit: per correction.  Hopefully the various indicated issues and corrections will be incorporated in upcoming changes to the site.

The changes appeared to be discussed on the discord.  Looks like there will be an effort for VPM to have a bit more of a crates.io style look or at least take some hints from it.

**Reply by i582** [reply]
> According to the [changelog](https://github.com/vlang/v/blob/master/CHANGELOG.md), there will be a new VPM site and hopefully the issues will be addressed.

No, this is a new site already :)

**Reply by i582** [reply]
> which is clearly gaining ground and maturing.

Interesting thought. Based on what data are you drawing these conclusions?

**Reply by Wajinn** [reply]
> No, this is a new site already :)

Yes, that's right.  That the necessary corrections get eventually made, is the important thing, and that users have the functionality.

> Interesting thought. Based on what data are you drawing these conclusions?

From when I first became aware of the project, until now, there is clear progression.  I've talked and corresponded with others about it, who think so too.  Of course, maybe others have different opinions, for their reasons.  Still, most are likely looking forward, as the project heads towards 0.3.5.

**Reply by medvednikov** [reply]
@Wajinn it's 0.4 this week actually :)

**Comment by i582** [comment]
Thanks for bringing this topic up! Indeed, I personally think that we should take the capabilities of other package managers in VPM, this will be enough to be convenient and useful.

However, right now we simply don't have developers to do it, some are busy with other tasks, some won't be able to make such a big project on their own, and some think that VPM doesn't deserve their time.

Perhaps by the autumn VPM will become part of the [VOSCA web projects](https://github.com/vlang-association), but so far without promises.

**Comment by ffred** [comment]
thanks for the replies !
of course I know that V is younger than Rust and couldn't compare with the organization behind Rust. so I understand that VPM will still need time to come closer to crates.io or others, for example like Nimble (for Nim lang). but I search in the forum and didn't found anyone saying anything about it, so just wanted to point something that I think is really important for a programming language now.

**Reply by i582** [reply]
> but I search in the forum and didn't found anyone saying anything about it

Most of the discussions takes place in the discord: https://discord.gg/vlang

**Comment by ffred** [comment]
ok, will go more on Discord..
